tests/integration.rs

```diff
@@ -0,0 +1,123 @@
+#[macro_use]
+mod utils;
+
+use std::{iter::once, path::PathBuf};
+
+use fs_err as fs;
+use parse_display::Display;
+use proptest::sample::size_range;
+use rand::{rngs::SmallRng, RngCore, SeedableRng};
+use tempfile::tempdir;
+use test_strategy::{proptest, Arbitrary};
+
+use crate::utils::{assert_same_directory, write_random_content};
+
+// tar and zip extensions
+#[derive(Arbitrary, Debug, Display)]
+#[display(style = "lowercase")]
+enum DirectoryExtension {
+    Tar,
+    Tbz,
+    Tgz,
+    Tlz4,
+    Tlz,
+    Tlzma,
+    Txz,
+    Zip,
+}
+
+// extensions of single file compression formats
+#[derive(Arbitrary, Debug, Display)]
+#[display(style = "lowercase")]
+enum FileExtension {
+    Bz,
+    Bz2,
+    Gz,
+    // Lz4, // broken
```

Comment:
Do you have any clues of what is causing this one? Maybe Lz4 would need to be filtered out in the next release?

Reply:
in normal cases its fine, but some formats are failing (`tar.lz4.lz4`, `tar.bz.lz4`)
It might have something to do with the library ([lz4_flex](https://lib.rs/crates/lz4_flex)), we can consider switching to liblz4 bindings like [lz4](https://lib.rs/crates/lz4) or [lzzzz](https://lib.rs/crates/lzzzz)

Reply:
Can you give me something to reproduce? Compressing compressed data should work

Reply:
@PSeitz if you prefer I can apply that on a branch for you to take a look at.

Reply:
Thanks I already applied it and try to extract the data which is failing

Reply:
I fixed the issues and created a PR here: https://github.com/ouch-org/ouch/pull/445